src/routes/internal/client/profiles.rs

```diff
@@ -0,0 +1,1292 @@
+use crate::auth::checks::filter_visible_version_ids;
+use crate::auth::{get_user_from_headers, AuthenticationError};
+use crate::database::models::legacy_loader_fields::MinecraftGameVersion;
+use crate::database::models::{
+    client_profile_item, generate_client_profile_id, generate_client_profile_link_id, version_item,
+};
+use crate::database::redis::RedisPool;
+use crate::file_hosting::FileHost;
+use crate::models::client::profile::{
+    ClientProfile, ClientProfileId, ClientProfileLinkId, ClientProfileShareLink,
+};
+use crate::models::ids::base62_impl::parse_base62;
+use crate::models::ids::{UserId, VersionId};
+use crate::models::pats::Scopes;
+use crate::queue::session::AuthQueue;
+use crate::routes::v3::project_creation::CreateError;
+use crate::routes::ApiError;
+use crate::util::routes::{read_from_field, read_from_payload};
+use crate::util::validate::validation_errors_to_string;
+use crate::{database, models};
+use actix_multipart::{Field, Multipart};
+use actix_web::web::Data;
+use actix_web::{web, HttpRequest, HttpResponse};
+use chrono::Utc;
+use futures::StreamExt;
+use itertools::Itertools;
+use rand::distributions::Alphanumeric;
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+use serde::{Deserialize, Serialize};
+use sha2::Digest;
+use sqlx::PgPool;
+use std::path::PathBuf;
+use std::sync::Arc;
+use validator::Validate;
+
+pub fn config(cfg: &mut web::ServiceConfig) {
+    cfg.service(
+        web::scope("client")
+            .route("profile", web::post().to(profile_create))
+            .route("check_token", web::get().to(profile_token_check))
+            .service(
+                web::scope("profile")
+                    .route("{id}", web::get().to(profile_get))
+                    .route("{id}", web::patch().to(profile_edit))
+                    .route("{id}", web::delete().to(profile_delete))
+                    .route("{id}/override", web::post().to(client_profile_add_override))
+                    .route(
+                        "{id}/override",
+                        web::delete().to(client_profile_remove_overrides),
+                    )
+                    .route("{id}/share", web::get().to(profile_share))
```

Comment:
IMO this route is kind of pointless. We should just return share links in the main profile body. When would you get a profile and not get its share links?